app/javascript/components/time_table/projects/external_auth.js

```diff
@@ -0,0 +1,147 @@
+  }
+
+  getProject () {
+    if (this.state.projectId) {
+      Api.makeGetRequest({ url: `/api/projects/${this.state.projectId}/external_auth` })
+         .then((response) => {
+           const { project, auth, public_key} = response.data;
+           if (!project.leader_id) project.leader_id = undefined;
+
+           this.setState({ project, auth, publicKey: public_key });
+         })
+    }
+  }
+
+  getAuthLink () {
+      Api.makeGetRequest({ url: `/api/external_auths/new?domain=${encodeURI(this.state.domain)}&provider=jira` })
+        .then((response) => {
+          const { authorization_url, request_data } = response.data;
+          this.setState({ authorizationUrl: authorization_url, requestData: request_data });
+        })
+  }
+
+  onDomainChange (e) {
+    const domain = e.target.value;
+    this.setState({ domain })
+  }
+
+  onTokenChange (e) {
+    const token = e.target.value;
+    this.setState({ token })
+  }
+
+  onSubmit (e) {
+    e.preventDefault()
+    const { domain, token } = this.state;
+    const authorization_url = this.state.authorizationUrl;
+    const request_data = this.state.requestData;
+    const project_id = this.state.projectId;
+    const provider = 'jira';
+    Api.makePostRequest({ url: "/api/external_auths", body: { external_auth: { domain, token, authorization_url, provider, request_data, project_id } } })
+        .then((response) => {
+          this.setState({ auth: response.data })
+        }).catch((error) => {
+          alert("Error during authorization")
+        })
+
+  }
+
+  onDelete (e) {
+    e.preventDefault();
+    Api.makeDeleteRequest({ url: `/api/external_auths/${this.state.auth.id}`})
+        .then(() => {
+          this.setState({ auth: undefined })
+        })
+  }
+
+  render () {
+    const { redirectToReferer, projectId } = this.state;
+
+    if (redirectToReferer) return <Redirect to={redirectToReferer} />
+    return (
+      <div>
+        {this._renderPublicKey()}
+        {this._renderAuth()}
+        <NavLink className="btn btn-primary" to={`/projects/${projectId}/edit`}>{I18n.t('common.cancel')}</NavLink>
+      </div>
+    )
+  }
+
+  _renderPublicKey () {
+    if (!this.state.publicKey || this.state.auth)
+      return null;
+    return (
+      [
+        <p key={0}>Public key:</p>,
+        <pre key={1}>
+          {this.state.publicKey}
+        </pre>
+      ]
+    );
+  }
+
+  _renderAuth () {
+    if (this.state.auth) {
+      return (
+        <div>
+          <p>
+             {`${I18n.t('apps.external_auths.authorized')} ${this.state.auth.provider}`}
+          </p>
+          <p>
+            <a className="btn btn-danger" onClick={this.onDelete}>{I18n.t('common.destroy')}</a>
+          </p>
+        </div>
+      );
+    }
+    return (
+      <div>
+        <h3>New auth</h3>
+        <input placeholder="Domain" value={this.state.domain} onChange={this.onDomainChange}></input>
+        <button onClick={this.getAuthLink}>Generate link</button>
```

Comment:
Missing translations. Please review all views and add missing translations.

Reply:
Fixed